### ISSUE TITLE
base maximum memory size on how much memory the Rumprun guest has, not a hardcoded limit

### ISSUE THREAD

**Issue by eyberg** [deferpanic/gorump#14]

**Comment by dstrbad** [comment]
Hi,

I'd be interested in this one, can you provide more details on it?

Thanks

**Comment by eyberg** [comment]
Since go requests a large chunk of memory off the bat it doesn't fit perfectly in-line with unikernel architectural goals (eg: being low on resources) so the size of memory used is fixed

https://github.com/deferpanic/gorump/commit/87eaedcdaf7d8d07f049dc0fe08edb9d1db0d8d6
https://github.com/deferpanic/gorump/commit/1d1d82be0a9bfc7bdfd396edbfdf1f39debff173

I don't know exactly what is best here to be honest - you prob. have a fair idea of what an app would cost going in so maybe configuration based but at the same time it'd be great to be able to have the least amount of disturbance w/in the tree as to not completely fork go

**Comment by anttikantee** [comment]
I don't think the memory allocation scheme can really be changed, but the amount of memory that the Rumprun instance has (e.g. -m parameter to qemu) should be reflected in the size of the chunk that Go reserves.  So essentially `mem_rumprun.go` should ask Rumprun how much memory is available.  However, the true memory size is currently not available via `sysctl`, so it requires some plumbing or magic spells.

**Comment by dstrbad** [comment]
Okay, so that essentially means that from host OS you don't know how much of reserved memory for rumpkernel is actually in use by guest process?

**Comment by anttikantee** [comment]
No. It means that Gorump does not know how much memory the guest can maximally allocate.  The host OS does not know a single thing about the memory usage. There might not even be a host OS -- gorump should also run directly on hardware.
